pkg/apis/apps/types.go

```diff
@@ -82,8 +82,8 @@ const (
 	// strategy, new Pods will be created from the specification version indicated
 	// by the StatefulSet's updateRevision.
 	RollingUpdateStatefulSetStrategyType StatefulSetUpdateStrategyType = "RollingUpdate"
-	// OnDeleteStatefulSetStrategyType triggers the legacy behavior. Version
-	// tracking and ordered rolling restarts are disabled. Pods are recreated
+	// OnDeleteStatefulSetStrategyType triggers most of the legacy behavior. Ordered
```

Comment:
This sentence sounds uncertain and is not great to read without mentioning any specifics.

Since most of the fields were not present in legacy behaviour (https://github.com/kubernetes/kubernetes/commit/1b55f57391f5a648b1f9859118a2d1fef7017ffe#diff-4607db7c58a5ac7a5e4f20c1e2b690c3b3416daafd7b913cda912f16faafef13L121) in my opinion there should not be any hindrance in backwards compatibility except having new updated fields.